src/main/java/org/elasticsearch/index/analysis/keyboard/KeyboardAnalyzerProvider.java

```diff
@@ -24,15 +24,18 @@
 
 public class KeyboardAnalyzerProvider extends AbstractIndexAnalyzerProvider<WhitespaceAnalyzer> {
 
-    private static final WhitespaceAnalyzer INSTANCE = new WhitespaceAnalyzer();
+    private final WhitespaceAnalyzer analyzer;
 
     public KeyboardAnalyzerProvider(IndexSettings indexSettings, Environment env,
                                     String name, Settings settings) {
         super(indexSettings, name, settings);
+
+        analyzer = new WhitespaceAnalyzer();
+        analyzer.setVersion(version);
     }
 
     @Override
     public WhitespaceAnalyzer get() {
-        return INSTANCE;
+        return this.analyzer;
```

Comment:
@aaerofeev Hi Could you please explain this change? Have you faced any issues with this analyzer at runtime?

Reply:
Hey. If the analyzer was not created in the constructor, I received an error when loading the plugin. I can’t provide the error text. But the point was that the analyzer was created too early.

I looked at this code in the example of creating an analyzer from the elasticsearch repository.

Reply:
Error similar like this
```
Analyzer is closed